### PR TITLE
[addons] refactor cstring handling in addon info builder

### PR DESCRIPTION
**Pull request by howie-f**
## Description
introduces `std::string StringUtils::StringOrEmpty(const char*)` which returns an empty string if `nullptr` is passed in. suggestions for a better naming are welcome. addon info builder was refactored afterwards to use this method.

replaced `vector.push_back()` with `.emplace_back()`

## How has this been tested?
about a week of live testing. nothing bad spotted

## What is the effect on users?
none
## Types of change

- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document

@AlwinEsch you may want to test it

### CODE REVIEW

**Review by AlwinEsch — APPROVED** [review]
From test it was OK and your changes makes it cleaner.

Never liked the "const char* cstring" brought by me, thanks for change it!

File: xbmc/utils/StringUtils.h [hunk 1]
```diff
@@ -375,6 +375,13 @@ class StringUtils
    */
   static std::string FormatFileSize(uint64_t bytes);
 
+  /*! \brief Converts a cstring pointer (const char*) to a std::string.
+             In case nullptr is passed the result is an empty string.
+      \param cstr the const pointer to char
+      \return the resulting std::string or ""
+   */
+  static std::string StringOrEmpty(const char* cstr);
```
Comment (ksooo) [inline xbmc/utils/StringUtils.h:383]: "Converts a cstring pointer (const char*) to a std::string"

I would name the new helper function `CreateFromCString" as this is what it does.

File: xbmc/addons/addoninfo/AddonInfoBuilder.cpp [hunk 2]
```diff
@@ -483,62 +471,59 @@ bool CAddonInfoBuilder::ParseXMLExtension(CAddonExtensions& addonExt, const TiXm
...
     {
-      cstring = attribute->Value();
-      if (cstring)
+      const std::string& value = StringUtils::StringOrEmpty(attribute->Value());
+      if (!value.empty())
       {
-        std::string value = cstring;
         name = "@" + name;
-        extension.push_back(std::make_pair(name, SExtValue(value)));
+        extension.emplace_back(std::make_pair(name, SExtValue(value)));
```
Comment (ksooo) [inline xbmc/addons/addoninfo/AddonInfoBuilder.cpp:478]: Those push_back to emplace_back changes are ofc okay, but either this change should be mentioned in the commit message or (better) put into a separate commit.

File: xbmc/addons/addoninfo/AddonInfoBuilder.cpp [hunk 3]
```diff
@@ -161,8 +156,8 @@ bool CAddonInfoBuilder::ParseXML(const AddonInfoPtr& addon, const TiXmlElement*
   const TiXmlElement* backwards = element->FirstChildElement("backwards-compatibility");
   if (backwards)
   {
-    cstring = backwards->Attribute("abi");
-    addon->m_minversion = AddonVersion(cstring ? cstring : "");
+    const std::string& minVersion = StringUtils::StringOrEmpty(backwards->Attribute("abi"));
```
Comment (ksooo) [inline xbmc/addons/addoninfo/AddonInfoBuilder.cpp:159]: I know compilers are quite clever these days, but as the new stringutils function returns an instance, not a ref, I would make that explicit by assigning the return value to a const std::string instance, not a const ref. The compiler will take care about return type optimization to avoid copies.

Comment (howie-f) [reply xbmc/addons/addoninfo/AddonInfoBuilder.cpp]: makes sense, applied your suggestions with the latest push

**Review by ksooo — APPROVED** [review]
lgtm
